### PR TITLE
SBERDOMA-996 tests and access to ticket

### PR DESCRIPTION
**Pull request by pahaz**
1. Resident user can not connect organization and property to ticket - so we need to give him access at least to id field
2. Resident user can not read all fields from Organization and Property - so we add access to those fields
3. Added tests to check if user can create and update Ticket and TicketFile

### CODE REVIEW

File: apps/condo/domains/property/access/Property.js [hunk 1]
```diff
@@ -7,11 +7,27 @@ const { queryOrganizationEmployeeFor } = require('@condo/domains/organization/ut
...
+            return false
+        }
+        const properties = residents.map(resident => get(resident, ['property', 'id']))
+        if (properties.length > 0) {
+            return {
+                id_in: properties,
+            }
+        }
+        return false
+    }
```
Comment (toplenboren) [inline apps/condo/domains/property/access/Property.js:30]: has duplicate in Organization access check

File: apps/condo/domains/property/schema/Property.js [hunk 2]
```diff
@@ -34,6 +35,7 @@ const Property = new GQLListSchema('Property', {
             schemaDoc: 'Client understandable Property name. A well-known property name for the client',
             type: Text,
             isRequired: false,
+            acess: userAccess.canAccessToStaffUserField,
```
Comment (toplenboren) [inline apps/condo/domains/property/schema/Property.js:38]:
```suggestion
            access: userAccess.canAccessToStaffUserField,
```

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/property/access/Property.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/property/access/Property.js:30]: we need to give access to property and to organization they are not duplicates. They both check access through Resident model

File: apps/condo/domains/property/schema/Property.js (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/property/schema/Property.js]: Fixed

File: apps/condo/domains/ticket/access/Ticket.js [hunk 3]
```diff
@@ -42,7 +47,10 @@ async function canManageTickets ({ authentication: { item: user }, operation, it
         if (!property) {
             return false
         }
-
+        // TODO(zuch): need to check organization and property
```
Comment (SavelevMatthew) [inline apps/condo/domains/ticket/access/Ticket.js:50]: ?)

**Review by SavelevMatthew — APPROVED** [review]
LGTM, but don't forget to check undone todos and fix resident tests also :)

File: apps/condo/domains/property/access/Property.js [hunk 4]
```diff
@@ -7,11 +7,27 @@ const { queryOrganizationEmployeeFor } = require('@condo/domains/organization/ut
...
     const userId = user.id
+    if (user.type === RESIDENT) {
+        const residents = await ResidentServerUtils.getAll(context, { user: { id: userId } })
+        if (residents.length === 0) {
+            return false
+        }
+        const properties = residents.map(resident => get(resident, ['property', 'id']))
+        if (properties.length > 0) {
+            return {
+                id_in: properties,
```
Comment (AntonAL) [inline apps/condo/domains/property/access/Property.js:26]: When user has several residents in one property, we will have duplicated ids here.
Use `uniq` from lodash:
```suggestion
                id_in: uniq(properties),
```

File: apps/condo/domains/organization/access/Organization.js [hunk 5]
```diff
@@ -2,13 +2,28 @@
...
     const userId = user.id
+    if (user.type === RESIDENT) {
+        const residents = await ResidentServerUtils.getAll(context, { user: { id: userId } })
+        if (residents.length === 0) {
+            return false
+        }
+        const organizations = residents.map(resident => get(resident, ['organization', 'id']))
+        if (organizations.length > 0) {
+            return {
+                id_in: organizations,
```
Comment (AntonAL) [inline apps/condo/domains/organization/access/Organization.js:22]: When user has several residents, being served with one organization, we will have duplicated ids here.
Use uniq from lodash:
```suggestion
                id_in: uniq(organizations),
```

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/ticket/access/Ticket.js (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/access/Ticket.js:50]: In fact we do not need to check this on create. If user do not have access to property or organization - he can not add such ticket. There will be connection error

File: apps/condo/domains/property/access/Property.js (hunk 4 above)
Comment (dkoviazin) [reply apps/condo/domains/property/access/Property.js]: done

**Pull request merged**